ddtrace/profiling/exporter/pprof.py

```diff
@@ -138,6 +139,21 @@ def convert_stack_event(self, thread_id, thread_native_id, thread_name, trace_id
         self._location_values[location_key]["cpu-time"] = sum(s.cpu_time_ns for s in samples)
         self._location_values[location_key]["wall-time"] = sum(s.wall_time_ns for s in samples)
 
+    def convert_memalloc_event(self, thread_id, thread_native_id, thread_name, frames, nframes, events):
+        location_key = (
+            self._to_locations(frames, nframes),
+            (("thread id", str(thread_id)), ("thread native id", str(thread_native_id)), ("thread name", thread_name),),
+        )
+
+        nevents = len(events)
+        sampling_ratio_avg = sum(event.capture_pct for event in events) / nevents / 100.0
```

Comment:
We could shave off some effort by collapsing these three iterations over events into one.

Reply:
You'd think so…

```python
l = list(range(10_000_000))

def v1():
    return sum(l), sum(l), sum(l)

def v2():
    x = y = z = 0
    for v in l:
         x += v
         y += v
         z += v
    return x, y, z


assert v1() == v2()


import timeit

print(timeit.timeit("v1()", number=10, globals=globals()))
print(timeit.timeit("v2()", number=10, globals=globals()))
```

v2 is 6 times slower:
```
$ python3 t.py
2.568303665
14.879553143
```

So I'm not sure it'd be a win even here 🤔

Reply:
Should have actually tested out my suggestion before making it! 🔬 

Funny but my initial thought was to use `functools.reduce` but that's even slower:

```
def v3():
    return functools.reduce(lambda x,y: (x[0]+y, x[1]+y, x[2]+y), l, (0,0,0))
```

Which takes ~21s.